### PR TITLE
Additional work during construction max price calculation

### PR DESCRIPTION
**Pull request by ranta**
# Hitas Pull Request

# Description

- Move `Rakennusaikaiset lisä- ja muutostyöt` from apartment improvements to apartment price calculation breakdown in confirmed maximum price calculations for pre 2011 cases
- Also some minor bugfixes

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
    - [x] Changes have been tested
- **Backend**
    - [x] Changes have been tested
    - [x] Automatic tests has been added
    - [x] OpenAPI definitions have been updated
    - [x] Database migrations will work in test environment
    - [ ] Oracle migration has been updated
    - [ ] initial.json has been updated to work with migrations

## Test plan

Create new maximum price calculation for a pre-2011 apartment, and verify that `Rakennusaikaiset lisä- ja muutostyöt` are not an improvement, and is listed in apartment breakdown
- [ ] Construction Price Index
- [ ] Market Price Index

## Tickets

This pull request resolves all or part of the following ticket(s): HT-353

### CODE REVIEW

File: backend/openapi.yaml [hunk 1]
```diff
@@ -3682,6 +3686,12 @@ components:
           type: number
           minimum: 0
           example: 1802.88
+        additional_work_during_construction:
+          description: Amount of apartment's additional work done during construction
+          type: number
+          nullable: true
```
Comment (matti-lamppu) [inline backend/openapi.yaml:3692]: Check if this is nullable, not maked so in the dataclass.

File: backend/openapi.yaml [hunk 2]
```diff
@@ -3778,6 +3790,18 @@ components:
...
```
Comment (matti-lamppu) [inline backend/openapi.yaml:3796]: Check if this is nullable, not maked so in the dataclass.

File: backend/openapi.yaml [hunk 3]
```diff
@@ -3778,6 +3790,18 @@ components:
...
+          minimum: 0
+          example: 0.0
+        index_adjusted_additional_work_during_construction:
```
Comment (matti-lamppu) [inline backend/openapi.yaml:3799]: Check if this is nullable, not maked so in the dataclass.

File: backend/openapi.yaml [hunk 4]
```diff
@@ -3778,6 +3790,18 @@ components:
...
+          description: Amount of apartment's additional work done during construction, after index adjustment
+          type: number
+          nullable: true
```
Comment (matti-lamppu) [inline backend/openapi.yaml:3802]: Check if this is nullable, not maked so in the dataclass.

File: frontend/src/app/services.ts [hunk 5]
```diff
@@ -53,7 +53,7 @@ const handleDownloadPDF = (response) => {
 
         const alink = document.createElement("a");
         alink.href = window.URL.createObjectURL(blob);
-        alink.download = `${filename}.pdf`;
+        alink.download = `${filename}`;
```
Comment (matti-lamppu) [inline frontend/src/app/services.ts:56]: ?

**Review by matti-lamppu — CHANGES_REQUESTED** [review]
Some things to double-check in the OpenAPI definition but otherwise looks good.

File: frontend/src/app/services.ts (hunk 5 above)
Comment (ranta) [reply frontend/src/app/services.ts:56]: It is explained in the commit message :)
```
frontend: Fix downloaded files containing duplicate filetype
Downloaded files were named e.g. `file.pdf.pdf`
```

File: backend/openapi.yaml (hunk 4 above)
Comment (ranta) [reply backend/openapi.yaml]: Good catch, fixed :)

**Review by matti-lamppu — APPROVED** [review]
👍

**Pull request merged**